scripts/experiments_tabular/maf.jl

```diff
@@ -0,0 +1,87 @@
+using DrWatson
+@quickactivate
+using ArgParse
+using GenerativeAD
+using StatsBase: fit!, predict, sample
+using BSON
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+   "max_seed"
+        required = true
+        arg_type = Int
+        help = "seed"
+    "dataset"
+        required = true
+        arg_type = String
+        help = "dataset"
+end
+parsed_args = parse_args(ARGS, s)
+@unpack dataset, max_seed = parsed_args
+
+modelname = "MAF"
+function sample_params()
+	par_vec = ([2, 5, 10], [10, 50, 100, 200, 500, 1000], 2:3, ["natural", "random"], [1f-4], [100], [30], [1f-6])
+	argnames = (:nflows, :hsize, :nlayers, :ordering, :lr, :batchsize, :patience, :wreg)
+
+	return (;zip(argnames, map(x->sample(x, 1)[1], par_vec))...)
+end
+
+function fit(data, parameters)
+	D = size(data[1][1], 1)
+
+	model = GenerativeAD.Models.MAF(
+				parameters.nflows,
+				D,
+				parameters.hsize,
+				parameters.nlayers,
+				parameters.ordering)
+
+	try
+		global info, fit_t, _, _, _ = @timed fit!(model, data, parameters)
+ 	catch e
+		@info "Failed training due to \n$e"
+		return (fit_t = NaN,), []
+	end
+
+	training_info = (
+		fit_t = fit_t,
+		history = info.history,
+		model = nothing
+		)
+
+	training_info, [(x -> predict(info.model, x), parameters)]
```

Comment:
Do you need to do `info.model`? Is not the original `model` variable changed during `fit!`? I think it is passed as reference.

Reply:
That's the dirty checkpoint trick. I need to deepcopy the model, in order to have some backup. I don't know how I can assign the model back to it's original reference.

Reply:
You could copy the parameters from the last working version of train model to the original variable. But this works.